src/System.Management.Automation/engine/debugger/Breakpoint.cs

```diff
@@ -482,9 +482,6 @@ internal bool TrySetBreakpoint(string scriptFile, FunctionContext functionContex
         {
             Diagnostics.Assert(SequencePointIndex == -1, "shouldn't be trying to set on a pending breakpoint");
 
-            if (!scriptFile.Equals(this.Script, StringComparison.OrdinalIgnoreCase))
```

Comment:
This is only called from a single place after we took them out from collection that linked to the current file. So we are sure those belong to the current file. So this is just unnecessary overhead.